edgelet/api/identityServiceVersion_2020_06_01.yaml

```diff
@@ -0,0 +1,102 @@
+swagger: '2.0'
+schemes:
+  - http
```

Comment:
What's the reason not using https?

Reply:
I believe this is because we don't do TLS over UDS. It is simply HTTP over UDS with peer cred auth. @arsing can correct me if I am wrong here.

Reply:
Yes, there's no point in doing TLS over UDS.

The only time TLS comes into play is for Kubernetes, because IS/KS/CS will have to bind to TCP, just like iotedged does today for the management and workload servers. But I don't think we need to represent that here in the spec. The specs are used for generating clients, and the clients don't care whether the underlying transport is UDS or TLS-over-TCP or whatever. The existing management and workload specs also just use `http`.

Reply:
It doesn't seem to me a valid reason. 

First of all, ADU doesn't plan to use UDS. Secondly we are assuming processes run on the same host, which is not the case in distributed scenarios: cluster or nested. For example, a child edge asks its parent for local identity. 

From client's perspective, it does need to know if the API server supports http only or http/https. And the schemes is for that purpose.

Reply:
>First of all, ADU doesn't plan to use UDS.

I don't understand. They don't have a choice but to use UDS.

>Secondly we are assuming processes run on the same host, which is not the case in distributed scenarios: cluster or nested. For example, a child edge asks its parent for local identity.

I covered the Kubernetes cluster case in my comment. For nested edge, I'm not sure downstream devices are going to connect to upstream device's IS? I've only heard of them connecting to upstream device's Edge Hub.

>From client's perspective, it does need to know if the API server supports http only or http/https.

It doesn't matter in the clients that we generate from the spec. As I said, we already have the management and workload swagger specs, and they only use `http`, which makes no difference because the clients are generic over the transport. We can add `https` here if you insist but it's not going to make any difference to the generated clients code.

Reply:
As @arsing said, UDS is the only available method of accessing this API. And we have been following the same pattern (and we don't plan to break that for now) in the new iotedged APIs.